client/components/tinymce/plugins/wpcom-view/plugin.js

```diff
@@ -321,7 +321,13 @@ function wpview( editor ) {
 		if ( pastedStr ) {
 			pastedStr = tinymce.trim( pastedStr.replace( /<[^>]+>/g, '' ) );
 
-			if ( /^https?:\/\/\S+$/i.test( pastedStr ) ) {
+			let imageMatch;
```

Comment:
Minor nitpick:

>When possible, variables should be declared using a const declaration. ([#](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#variable-declarations))

Although what you have works, we can avoid the `let` by defining:
`const imageMatch = /(https?:\/\/[^<]*)(\.jpg|\.jpeg|\.gif|\.png)\??.*$/i.exec( pastedStr );`

Other than that, tested and LGTM. Thanks for the quick fix!